util/archutil/detect.go

```diff
@@ -4,21 +4,28 @@ import (
 	"sort"
 	"strings"
 	"sync"
+	"time"
 
 	"github.com/containerd/containerd/platforms"
 	"github.com/moby/buildkit/util/bklog"
 	ocispecs "github.com/opencontainers/image-spec/specs-go/v1"
 )
 
-var mu sync.Mutex
-var arr []ocispecs.Platform
+var CacheMaxAge = 20 * time.Second
```

Comment:
Out of curiosity. How often does this change during the lifetime of a buildkit process? I guess it would only change on an install of new binfmt handlers right?

Potentially, could we invalidate the cache based on the contents of `/proc/sys/fs/binfmt_misc/`? Just a thought, curious if you considered this option.